merkledag/cidversion.go

```diff
@@ -0,0 +1,26 @@
+package merkledag
+
+import (
+	"fmt"
+)
+
+// CidVersion is a a cid version number that is guaranteed to be valid
+type CidVersion struct {
```

Comment:
Also note, I am not against doing the way you suggested.

I just want to be able to assume that once we have a CidVersion we can guarantee we have something valid and don't want to have to return an error (instead we can panic).  I think it will be reasonable to assume that the user (in this case the programmer) won't do a cast  from an int to to CidVersion and will use the helper function.

@whyrusleeping thoughts.